src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfigurationTests.cs

```diff
@@ -0,0 +1,173 @@
+                // to all users on the machine e.g. /etc/dotnet-monitor on Unix systems.
+                string sharedSettingsContent = JsonSerializer.Serialize(SharedSettingsContent);
+                File.WriteAllText(Path.Combine(sharedConfigDir.FullName, "settings.json"), sharedSettingsContent);
+            }
+            if (level >= ConfigurationLevel.SharedKeyPerFile)
+            {
+                // This is a key-per-file file in the shared configuration directory. This configuration
+                // is typically used when mounting secrets from a Docker volume.
+                File.WriteAllText(Path.Combine(sharedConfigDir.FullName, WebHostDefaults.ServerUrlsKey), nameof(ConfigurationLevel.SharedKeyPerFile));
+            }
+
+            // Create the initial host builder.
+            IHostBuilder builder = HostBuilderHelper.CreateHostBuilder(settings);
+
+            // Override the environment configurations to use predefined values so that the test host
+            // doesn't inadvertently provide unexpected values. Passing null replaces with an empty
+            // in-memory collection source.
+            builder.ReplaceAspnetEnvironment(level >= ConfigurationLevel.AspnetEnvironment ? AspnetEnvironmentVariables : null); ;
+            builder.ReplaceDotnetEnvironment(level >= ConfigurationLevel.DotnetEnviroment ? DotnetEnvironmentVariables : null);
+            builder.ReplaceMonitorEnvironment(level >= ConfigurationLevel.MonitorEnvironment ? MonitorEnvironmentVariables : null);
+
+            // Build the host and get the Urls property from configuration.
+            IHost host = builder.Build();
+            IConfiguration rootConfiguration = host.Services.GetRequiredService<IConfiguration>();
+            string configuredUrls = rootConfiguration[WebHostDefaults.ServerUrlsKey];
+
+            // Test that the value of the Urls property is the same as it was set
+            // for the level of configuration of the test.
+            if (level == ConfigurationLevel.None)
+            {
+                Assert.Null(configuredUrls);
+            }
+            else if (level == ConfigurationLevel.DotnetEnviroment)
+            {
+                // Not sure this was the intent that setting DOTNET_Urls via environment yields no
+                // configured URLS, however, this does not happen because the --urls command line
+                // parameter always supercedes this environment variable (again, not sure this was
+                // intended either).
+                Assert.Null(configuredUrls);
+            }
+            else
+            {
+                Assert.Equal(Enum.GetName(level), configuredUrls);
+            }
+        }
+
+        /// This is the order of configuration sources where a name with a lower
+        /// enum value has a lower precedence in configuration.
+        public enum ConfigurationLevel
```

Comment:
One point of confusion here - is `MonitorUrls` supposed to be equivalent to command line parameters? I expected a command line enum value to be roughly where `MonitorUrls` is, but I don't fully know what that means.

Reply:
In terms of thinking of this as the host builder for dotnet-monitor's `collect` and `config show` commands, yes. However, the is testing at the configuration building level, so it's saying that the Urls parameter of the settings is being set. Calling it `CommandLine` would be conflating it with the default configuration settings where command line arguments are inserted (which our host builder doesn't use).

Reply:
I've renamed it to `HostBuilderSettingsUrl` to make it clear what is being set.